lib/job.js

```diff
@@ -108,6 +109,103 @@ function _addTasksWithLen(info, db_entry, storage, len, split_size) {
  * @param  {object} client DB entry for this client
  * @return {object} The task information required by the client.
  */
-module.exports.schedule = function(client) {
+module.exports.schedule = function(client,storage,res) {
     // TODO schedule the job
+    /*var chain = new Sequelize.Utils.QueryChainer();
+    var prev_jobs = client.previous_job;
+    for(i in prev_jobs) {
+        chain.add(storage.Task.findOne({
+            where:{
+                taken: 0,
+                job_id: prev_jobs[i].job_id,
+                step:prev_jobs[i].step
+            }
+        }));
+    }
+    //If prev_jobs was empty or no task with same job and same step is available
+    if(chain.length < 1){
+       chain.add(storage.Task.findOne({
+           where:{
+               taken:0
+           }
+       }))
+    }
+
+    chain.run().success(function(){
+
+    })
+
+  */
+        storage.Task.findOne({
+            where: {
+                taken: 0
+            }
+             //todo add condition for correct stage
+        }).then(function (task) {
+            if (!task)
+                return;
```

Comment:
I think this will result in the client never getting a response (loading forever) if there is no task. probably need to call:
```js
res.json({
    registered: false
});
```
here too